toolkit/recordreplay/ProcessRecordReplay.cpp

```diff
@@ -873,6 +881,11 @@ void MaybeCreateCheckpoint() {
 static bool gTearingDown;
 
 void FinishRecording() {
+  // Ensure that the driver connected and created the recording, if it is
+  // configured to do so. Otherwise we will never notify the user that the
+  // recording is unusable.
+  gWaitForRecordingCreated();
```

Comment:
I don't understand this comment.  js::SendRecordingFinished reports that the recording is unusable if an unusable reason has been set, and that doesn't seem to depend on whether the recording has been created by the backend?

Reply:
The issue this aims to address is that no part of this process, from what I can see, will wait for the connection to the Replay servers have been opened, so it is possible for the connection to fail, and the recording to never be created.

This code aimed at stopping us from calling `js::SendRecordingFinished()` before the driver has opened the connection and created the recording. This is to ensure that when we do call `js::SendRecordingFinished()`, the driver will have had the opportunity to mark the recording as unusable, if necessary. This also ensures that when `js::SendRecordingFinished()` runs and notifies the parent, the parent can safely assume that API calls based on the `recordingId` will not return an error.

Reply:
OK, makes sense!  Can you update the comment to indicate that the recording may become unusable due to a connection problem?